src/athena/src/components/screens/auth/SignUp.tsx

```diff
@@ -0,0 +1,190 @@
+interface SignUpProps extends ParamList {}
+
+interface SignUpInput {
+  name: string;
+  username: string;
+  email: string;
+  password: string;
+}
+
+const initialValues: SignUpInput = {
+  name: '',
+  username: '',
+  email: '',
+  password: '',
+};
+
+export const SignUp: React.FC<SignUpProps> = ({ navigation }) => {
+  const [signup] = useSignUpMutation();
+  const [findusername] = useFindUsernameMutation();
+  const [findemail] = useFindEmailMutation();
+
+  const SignUpSchema = object({
+    name: string().required(),
+    username: string()
+      .required()
+      .test('', 'Username is not available.', async (username) => {
+        const { data } = await findusername({ variables: { input: username || '' } });
+
+        return !data?.findusername;
+      }),
+    email: string()
+      .required()
+      .email('Must provide valid email address.')
+      .test('', 'This email is already used on another account.', async (email) => {
+        const { data } = await findemail({ variables: { input: email || '' } });
+
+        return !data?.findemail;
+      }),
+    password: string()
+      .required()
+      .matches(
+        /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,20}$/,
+        'Password must be between 6 to 20 characters and contain at least 1 upper case letter, 1 lower case letter and 1 number.'
+      ),
+  });
+
+  const validate = async (values: SignUpInput) => {
+    const errors: Record<string, string> = {};
+    await SignUpSchema.validate(values, { abortEarly: false }).catch((err: ValidationError) => {
+      err.inner.forEach(({ path, errors: errorMessage }) => {
+        if (path && !errors[path]) {
+          errors[path] = errorMessage[0];
+        }
+      });
+    });
+    return errors;
+  };
+
+  const onSubmit = async (values: SignUpInput) => {
+    await signup({ variables: { input: values } });
+
+    navigation.navigate('Login');
+  };
+
+  return (
+    <SafeAreaView>
+      <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
+        <View style={styles.screen}>
+          <KeyboardAvoidingView style={styles.keyboardView} behavior={'padding'}>
+            <Text style={{ ...styles.heading2, marginBottom: 20 }}>Sign Up</Text>
+            <Formik initialValues={initialValues} onSubmit={onSubmit} validate={validate}>
+              {({ handleChange, handleBlur, handleSubmit, values, errors, touched }) => {
+                return (
+                  <>
+                    <Input
+                      placeholder="name"
+                      value={values.name}
+                      onChangeText={handleChange('name')}
+                      autoCorrect={false}
+                      autoCapitalize="none"
+                      error={touched.name && !!errors.name}
+                      onBlur={handleBlur('name')}
+                      style={{ marginBottom: 14 }}
+                    />
+                    {touched.name && !!errors.name && (
```

Comment:
```suggestion
                    {touched.name && errors.name && (
```
Nit: Unnecessary cast